include/rocksdb/filter_policy.h

```diff
@@ -139,6 +154,10 @@ struct FilterBuildingContext {
 
   // Reason for creating the file with the filter
   TableFileCreationReason reason = TableFileCreationReason::kMisc;
+
+  // Optional: if provided, it charges memory usage of bloom filter/ribbon
+  // filter construction to block cache
+  std::shared_ptr<CacheReservationManager> cache_res_mgr;
```

Comment:
From chat: The FilterBitsBuilder should be able to "self serve" by looking up options in the `table_options`. Thus, I don't think we need to expose CacheReservationManager